File: app/src/main/java/com/example/readative/room/entity/Shelf.java
package com.example.readative.room.entity;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class Shelf {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "shelf_id")
    public long id;

    @NonNull
    @ColumnInfo(name = "shelf_name")
    public String name;
    @NonNull
    @ColumnInfo(name = "shelf_hidden")
    public boolean isHidden;
}

File: domain/model/src/main/java/com/example/model/local/entity/Shelf.kt
package com.example.model.local.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "shelves",
    indices = [
        Index(value = ["name"], unique = true)
    ]
)
data class Shelf(
    @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "id") override val id: Long,
    @ColumnInfo(name = "name") val name: String,
    @ColumnInfo(name = "hidden_status") val isHidden: Boolean = false
) : ReadativeEntity
